network/p2p/scoring/registry_test.go

```diff
@@ -681,3 +803,8 @@ func penaltyValueFixture(msgType p2pmsg.ControlMessageType) float64 {
 		return penaltyValues.ClusterPrefixedPenaltyReductionFactor
 	}
 }
+
+func stopRegistry(t *testing.T, cancel context.CancelFunc, registry *scoring.GossipSubAppSpecificScoreRegistry) {
+	cancel()
+	unittest.RequireCloseBefore(t, registry.Done(), 500*time.Millisecond, "registry did not stop")
```

Comment:
```suggestion
	unittest.RequireCloseBefore(t, registry.Done(), 5*time.Second, "registry did not stop")
```
We may consider a more stretched startup and shutdown timeout, as the test can get flakey solely based on these timeouts on slow CI machines.